src/platform/bouffalolab/BL602/NetworkCommissioningDriver.cpp

```diff
@@ -208,27 +189,41 @@ void BLWiFiDriver::ConnectNetwork(ByteSpan networkId, ConnectCallback * callback
     }
 }
 
-CHIP_ERROR BLWiFiDriver::StartScanWiFiNetworks(ByteSpan ssid)
+void BLWiFiDriver::ScanNetworks(ByteSpan ssid, WiFiDriver::ScanCallback * callback)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
+    int iret = 0;
 
-    if (!ssid.empty())
-    {
-        memset(WiFiSSIDStr, 0, sizeof(WiFiSSIDStr));
-        memcpy(WiFiSSIDStr, ssid.data(), ssid.size());
-        err       = (CHIP_ERROR) wifi_mgmr_scan_adv(NULL, NULL, NULL, 0, NULL, WiFiSSIDStr, 1, 0);
-        scan_type = 1;
-    }
-    else
-    {
-        err       = (CHIP_ERROR) wifi_mgmr_scan(NULL, NULL);
-        scan_type = 0;
-    }
-    if (err != CHIP_NO_ERROR)
+    if (callback != nullptr)
     {
-        return CHIP_ERROR_INTERNAL;
+        if (!ssid.empty())
+        {
+            if (ssid.size() <= sizeof(mScanSSID))
+            {
+                memset(mScanSSID, 0, sizeof(mScanSSID));
+                memcpy(mScanSSID, ssid.data(), ssid.size());
```

Comment:
Again: what happens if `ssid.size() == sizeof(mScanSSID)`?  Please explain to me why that behavior is ok.

Reply:
Add one more byte on mScanSSID and only do memcpy when ssid.size() < sizeof(mScanSSID)